diff --git a/backend/driver.py b/backend/driver.py
index a4ac9ef..69f50b3 100644
--- a/backend/driver.py
+++ b/backend/driver.py
@@ -9,6 +9,7 @@ import subprocess
 import platform
 import importlib.util
 import sys
+import time
 
 from pathlib import Path
 
@@ -461,9 +462,56 @@ class CPUDriver(DriverBase):
     def is_active():
         return False
 
+    def do_bench(self, fn, warmup=25, rep=100, grad_to_none=None, quantiles=None, return_mode="mean"):
+        assert return_mode in ["min", "max", "mean", "median", "all"]
+        fn()
+
+        # cache = runtime.driver.active.get_empty_cache_for_benchmark()
+
+        # Estimate the runtime of the function
+        start_event = time.perf_counter()
+        for _ in range(5):
+            # runtime.driver.active.clear_cache(cache)
+            fn()
+        end_event = time.perf_counter()
+        estimate_ms = int((end_event - start_event)*1000 / 5)
+
+        from triton import knobs
+        verbose = knobs.autotuning.print
+        if verbose:
+            print("CPU estimate ms: ", estimate_ms)
+
+        # compute number of warmup and repeat
+        n_warmup = max(1, int(warmup / estimate_ms))
+        n_repeat = max(5, int(rep / estimate_ms))
+        start_event = [0 for i in range(n_repeat)]
+        end_event = [0 for i in range(n_repeat)]
+        # Warm-up
+        for _ in range(n_warmup):
+            fn()
+        # Benchmark
+        for i in range(n_repeat):
+            # we don't want `fn` to accumulate gradient values
+            # if it contains a backward pass. So we clear the
+            # provided gradients
+            if grad_to_none is not None:
+                for x in grad_to_none:
+                    x.grad = None
+            # we clear the L2 cache before each run
+            # runtime.driver.active.clear_cache(cache)
+            # record time of `fn`
+            start_event[i] = time.perf_counter()
+            fn()
+            end_event[i] = time.perf_counter()
+        times = [(e-s)*1000 for s, e in zip(start_event, end_event)]
+        if verbose:
+            print("KERNEL TIMES: ", ", ".join(str(t) for t in times))
+
+        from triton.testing import _summarize_statistics
+        return _summarize_statistics(times, quantiles, return_mode)
+
     def get_benchmarker(self):
-        from triton.testing import do_bench
-        return do_bench
+        return self.do_bench
 
     def get_device_capability(self):
         return ("cpu", 0)
diff --git a/triton-hash.txt b/triton-hash.txt
index 57a17d5..ca05e91 100644
--- a/triton-hash.txt
+++ b/triton-hash.txt
@@ -1 +1 @@
-32b7b208d48d2d06c8af2cbf1b45d8c4c70d7f1f
\ No newline at end of file
+32b7b208d48d2d06c8af2cbf1b45d8c4c70d7f1f
