diff --git a/backend/compiler.py b/backend/compiler.py
index dd6fc89..ee941bb 100644
--- a/backend/compiler.py
+++ b/backend/compiler.py
@@ -1,6 +1,3 @@
-# Copyright (c) Meta Platforms, Inc. and affiliates, Microsoft Corporation.
-# Licensed under the MIT license.
-
 from triton.backends.compiler import BaseBackend, GPUTarget
 from triton._C.libtriton import ir, passes
 from dataclasses import dataclass
diff --git a/backend/driver.py b/backend/driver.py
index a4ac9ef..de45f70 100644
--- a/backend/driver.py
+++ b/backend/driver.py
@@ -1,6 +1,3 @@
-# Copyright (c) Meta Platforms, Inc. and affiliates, Microsoft Corporation.
-# Licensed under the MIT license.
-
 import hashlib
 import tempfile
 
@@ -9,6 +6,7 @@ import subprocess
 import platform
 import importlib.util
 import sys
+import time
 
 from pathlib import Path
 
@@ -461,9 +459,56 @@ class CPUDriver(DriverBase):
     def is_active():
         return False
 
+    def do_bench(self, fn, warmup=25, rep=100, grad_to_none=None, quantiles=None, return_mode="mean"):
+        assert return_mode in ["min", "max", "mean", "median", "all"]
+        fn()
+
+        # cache = runtime.driver.active.get_empty_cache_for_benchmark()
+
+        # Estimate the runtime of the function
+        start_event = time.perf_counter()
+        for _ in range(5):
+            # runtime.driver.active.clear_cache(cache)
+            fn()
+        end_event = time.perf_counter()
+        estimate_ms = int((end_event - start_event)*1000 / 5)
+      
+        from triton import knobs
+        verbose = knobs.autotuning.print
+        if verbose:
+            print("CPU estimate ms: ", estimate_ms)
+
+        # compute number of warmup and repeat
+        n_warmup = max(1, int(warmup / estimate_ms))
+        n_repeat = max(5, int(rep / estimate_ms))
+        start_event = [0 for i in range(n_repeat)]
+        end_event = [0 for i in range(n_repeat)]
+        # Warm-up
+        for _ in range(n_warmup):
+            fn()
+        # Benchmark
+        for i in range(n_repeat):
+            # we don't want `fn` to accumulate gradient values
+            # if it contains a backward pass. So we clear the
+            # provided gradients
+            if grad_to_none is not None:
+                for x in grad_to_none:
+                    x.grad = None
+            # we clear the L2 cache before each run
+            # runtime.driver.active.clear_cache(cache)
+            # record time of `fn`
+            start_event[i] = time.perf_counter()
+            fn()
+            end_event[i] = time.perf_counter()
+        times = [(e-s)*1000 for s, e in zip(start_event, end_event)]
+        if verbose: 
+            print("KERNEL TIMES: ", ", ".join(str(t) for t in times))
+
+        from triton.testing import _summarize_statistics
+        return _summarize_statistics(times, quantiles, return_mode)
+
     def get_benchmarker(self):
-        from triton.testing import do_bench
-        return do_bench
+        return self.do_bench
 
     def get_device_capability(self):
         return ("cpu", 0)
diff --git a/include/triton-shared/Analysis/MaskAnalysis.h b/include/triton-shared/Analysis/MaskAnalysis.h
index 8ca8cc4..0ad7c6f 100644
--- a/include/triton-shared/Analysis/MaskAnalysis.h
+++ b/include/triton-shared/Analysis/MaskAnalysis.h
@@ -1,6 +1,6 @@
 //===----------------------------------------------------------------------===//
 //
-// Copyright (c) Meta Platforms, Inc. and affiliates, Microsoft Corporation.
+// Copyright (c) Microsoft Corporation, Meta Platforms.
 // Licensed under the MIT license.
 //
 //===----------------------------------------------------------------------===//
diff --git a/include/triton-shared/Analysis/OpFoldResultUtils.h b/include/triton-shared/Analysis/OpFoldResultUtils.h
index 476a8fd..248dfc3 100644
--- a/include/triton-shared/Analysis/OpFoldResultUtils.h
+++ b/include/triton-shared/Analysis/OpFoldResultUtils.h
@@ -1,6 +1,6 @@
 //===----------------------------------------------------------------------===//
 //
-// Copyright (c) Meta Platforms, Inc. and affiliates, Microsoft Corporation.
+// Copyright (c) Microsoft Corporation, Meta Platforms.
 // Licensed under the MIT license.
 //
 //===----------------------------------------------------------------------===//
diff --git a/include/triton-shared/Analysis/PtrAnalysis.h b/include/triton-shared/Analysis/PtrAnalysis.h
index 759cc78..5a95ebd 100644
--- a/include/triton-shared/Analysis/PtrAnalysis.h
+++ b/include/triton-shared/Analysis/PtrAnalysis.h
@@ -1,6 +1,6 @@
 //===----------------------------------------------------------------------===//
 //
-// Copyright (c) Meta Platforms, Inc. and affiliates, Microsoft Corporation.
+// Copyright (c) Microsoft Corporation.
 // Licensed under the MIT license.
 //
 //===----------------------------------------------------------------------===//
diff --git a/include/triton-shared/Analysis/UseAnalysis.h b/include/triton-shared/Analysis/UseAnalysis.h
index 65772b5..f3dcef3 100644
--- a/include/triton-shared/Analysis/UseAnalysis.h
+++ b/include/triton-shared/Analysis/UseAnalysis.h
@@ -1,6 +1,6 @@
 //===----------------------------------------------------------------------===//
 //
-// Copyright (c) Meta Platforms, Inc. and affiliates, Microsoft Corporation.
+// Copyright (c) Microsoft Corporation.
 // Licensed under the MIT license.
 //
 //===----------------------------------------------------------------------===//
diff --git a/include/triton-shared/AnalysisStructured/PtrAnalysis.h b/include/triton-shared/AnalysisStructured/PtrAnalysis.h
index 306a7b5..b1791aa 100644
--- a/include/triton-shared/AnalysisStructured/PtrAnalysis.h
+++ b/include/triton-shared/AnalysisStructured/PtrAnalysis.h
@@ -1,6 +1,6 @@
 //===----------------------------------------------------------------------===//
 //
-// Copyright (c) Meta Platforms, Inc. and affiliates, Microsoft Corporation.
+// Copyright (c) Microsoft Corporation, Meta Platforms.
 // Licensed under the MIT license.
 //
 //===----------------------------------------------------------------------===//
diff --git a/include/triton-shared/Conversion/StructuredToMemref/Passes.h b/include/triton-shared/Conversion/StructuredToMemref/Passes.h
index 1af4f3c..198675b 100644
--- a/include/triton-shared/Conversion/StructuredToMemref/Passes.h
+++ b/include/triton-shared/Conversion/StructuredToMemref/Passes.h
@@ -1,10 +1,3 @@
-//===----------------------------------------------------------------------===//
-//
-// Copyright (c) Meta Platforms, Inc. and affiliates, Microsoft Corporation.
-// Licensed under the MIT license.
-//
-//===----------------------------------------------------------------------===//
-
 #ifndef TRITON_STRUCTURED_TO_MEMREF_CONVERSION_PASSES_H
 #define TRITON_STRUCTURED_TO_MEMREF_CONVERSION_PASSES_H
 
diff --git a/include/triton-shared/Conversion/StructuredToMemref/Passes.td b/include/triton-shared/Conversion/StructuredToMemref/Passes.td
index be56471..ac318f5 100644
--- a/include/triton-shared/Conversion/StructuredToMemref/Passes.td
+++ b/include/triton-shared/Conversion/StructuredToMemref/Passes.td
@@ -1,10 +1,3 @@
-//===----------------------------------------------------------------------===//
-//
-// Copyright (c) Meta Platforms, Inc. and affiliates, Microsoft Corporation.
-// Licensed under the MIT license.
-//
-//===----------------------------------------------------------------------===//
-
 #ifndef STRUCTURED_TO_MEMREF_CONVERSION_PASSES
 #define STRUCTURED_TO_MEMREF_CONVERSION_PASSES
 
diff --git a/include/triton-shared/Conversion/StructuredToMemref/StructuredToMemref.h b/include/triton-shared/Conversion/StructuredToMemref/StructuredToMemref.h
index 196cc22..8c67c9e 100644
--- a/include/triton-shared/Conversion/StructuredToMemref/StructuredToMemref.h
+++ b/include/triton-shared/Conversion/StructuredToMemref/StructuredToMemref.h
@@ -1,10 +1,3 @@
-//===----------------------------------------------------------------------===//
-//
-// Copyright (c) Meta Platforms, Inc. and affiliates, Microsoft Corporation.
-// Licensed under the MIT license.
-//
-//===----------------------------------------------------------------------===//
-
 #ifndef TRITON_CONVERSION_STRUCTUREDTOMEMREF_STRUCTUREDTOMEMREF_H
 #define TRITON_CONVERSION_STRUCTUREDTOMEMREF_STRUCTUREDTOMEMREF_H
 
diff --git a/include/triton-shared/Conversion/TritonArithToLinalg/ConversionPatterns.hpp b/include/triton-shared/Conversion/TritonArithToLinalg/ConversionPatterns.hpp
index d1a76bf..c509be3 100644
--- a/include/triton-shared/Conversion/TritonArithToLinalg/ConversionPatterns.hpp
+++ b/include/triton-shared/Conversion/TritonArithToLinalg/ConversionPatterns.hpp
@@ -3,7 +3,7 @@
 
 //===----------------------------------------------------------------------===//
 //
-// Copyright (c) Meta Platforms, Inc. and affiliates, Microsoft Corporation.
+// Copyright (c) Microsoft Corporation, Meta Platforms.
 // Licensed under the MIT license.
 //
 //===----------------------------------------------------------------------===//
diff --git a/include/triton-shared/Conversion/TritonArithToLinalg/ConversionTools.h b/include/triton-shared/Conversion/TritonArithToLinalg/ConversionTools.h
index a0f5047..b3b42c1 100644
--- a/include/triton-shared/Conversion/TritonArithToLinalg/ConversionTools.h
+++ b/include/triton-shared/Conversion/TritonArithToLinalg/ConversionTools.h
@@ -1,10 +1,3 @@
-//===----------------------------------------------------------------------===//
-//
-// Copyright (c) Meta Platforms, Inc. and affiliates, Microsoft Corporation.
-// Licensed under the MIT license.
-//
-//===----------------------------------------------------------------------===//
-
 #ifndef TRITON_CONVERSION_TRITONARITHTOLINALG_CONVERSIONTOOLS_H
 #define TRITON_CONVERSION_TRITONARITHTOLINALG_CONVERSIONTOOLS_H
 
diff --git a/include/triton-shared/Conversion/TritonArithToLinalg/Passes.h b/include/triton-shared/Conversion/TritonArithToLinalg/Passes.h
index e748c4c..b95cbde 100644
--- a/include/triton-shared/Conversion/TritonArithToLinalg/Passes.h
+++ b/include/triton-shared/Conversion/TritonArithToLinalg/Passes.h
@@ -1,10 +1,3 @@
-//===----------------------------------------------------------------------===//
-//
-// Copyright (c) Meta Platforms, Inc. and affiliates, Microsoft Corporation.
-// Licensed under the MIT license.
-//
-//===----------------------------------------------------------------------===//
-
 #ifndef TRITON_ARITH_TO_LINALG_CONVERSION_PASSES_H
 #define TRITON_ARITH_TO_LINALG_CONVERSION_PASSES_H
 
diff --git a/include/triton-shared/Conversion/TritonArithToLinalg/Passes.td b/include/triton-shared/Conversion/TritonArithToLinalg/Passes.td
index 1a55303..d1f0f63 100644
--- a/include/triton-shared/Conversion/TritonArithToLinalg/Passes.td
+++ b/include/triton-shared/Conversion/TritonArithToLinalg/Passes.td
@@ -1,10 +1,3 @@
-//===----------------------------------------------------------------------===//
-//
-// Copyright (c) Meta Platforms, Inc. and affiliates, Microsoft Corporation.
-// Licensed under the MIT license.
-//
-//===----------------------------------------------------------------------===//
-
 #ifndef TRITON_ARITH_TO_LINALG_CONVERSION_PASSES
 #define TRITON_ARITH_TO_LINALG_CONVERSION_PASSES
 
diff --git a/include/triton-shared/Conversion/TritonArithToLinalg/TritonArithToLinalg.h b/include/triton-shared/Conversion/TritonArithToLinalg/TritonArithToLinalg.h
index 45867e4..22dc652 100644
--- a/include/triton-shared/Conversion/TritonArithToLinalg/TritonArithToLinalg.h
+++ b/include/triton-shared/Conversion/TritonArithToLinalg/TritonArithToLinalg.h
@@ -1,10 +1,3 @@
-//===----------------------------------------------------------------------===//
-//
-// Copyright (c) Meta Platforms, Inc. and affiliates, Microsoft Corporation.
-// Licensed under the MIT license.
-//
-//===----------------------------------------------------------------------===//
-
 #ifndef TRITON_CONVERSION_TRITONARITHTOLINALG_TRITONARITHTOLINALG_H
 #define TRITON_CONVERSION_TRITONARITHTOLINALG_TRITONARITHTOLINALG_H
 
diff --git a/include/triton-shared/Conversion/TritonPtrToMemref/Passes.h b/include/triton-shared/Conversion/TritonPtrToMemref/Passes.h
index f0df4ab..e1f6f33 100644
--- a/include/triton-shared/Conversion/TritonPtrToMemref/Passes.h
+++ b/include/triton-shared/Conversion/TritonPtrToMemref/Passes.h
@@ -1,10 +1,3 @@
-//===----------------------------------------------------------------------===//
-//
-// Copyright (c) Meta Platforms, Inc. and affiliates, Microsoft Corporation.
-// Licensed under the MIT license.
-//
-//===----------------------------------------------------------------------===//
-
 #ifndef TRITON_PTR_TO_MEMREF_CONVERSION_PASSES_H
 #define TRITON_PTR_TO_MEMREF_CONVERSION_PASSES_H
 
diff --git a/include/triton-shared/Conversion/TritonPtrToMemref/Passes.td b/include/triton-shared/Conversion/TritonPtrToMemref/Passes.td
index 0aaec4c..c027b09 100644
--- a/include/triton-shared/Conversion/TritonPtrToMemref/Passes.td
+++ b/include/triton-shared/Conversion/TritonPtrToMemref/Passes.td
@@ -1,10 +1,3 @@
-//===----------------------------------------------------------------------===//
-//
-// Copyright (c) Meta Platforms, Inc. and affiliates, Microsoft Corporation.
-// Licensed under the MIT license.
-//
-//===----------------------------------------------------------------------===//
-
 #ifndef TRITON_PTR_TO_MEMREF_CONVERSION_PASSES
 #define TRITON_PTR_TO_MEMREF_CONVERSION_PASSES
 
diff --git a/include/triton-shared/Conversion/TritonPtrToMemref/TritonPtrToMemref.h b/include/triton-shared/Conversion/TritonPtrToMemref/TritonPtrToMemref.h
index c2df315..4476f7d 100644
--- a/include/triton-shared/Conversion/TritonPtrToMemref/TritonPtrToMemref.h
+++ b/include/triton-shared/Conversion/TritonPtrToMemref/TritonPtrToMemref.h
@@ -1,10 +1,3 @@
-//===----------------------------------------------------------------------===//
-//
-// Copyright (c) Meta Platforms, Inc. and affiliates, Microsoft Corporation.
-// Licensed under the MIT license.
-//
-//===----------------------------------------------------------------------===//
-
 #ifndef TRITON_CONVERSION_TRITON_PTR_TO_MEMREF_TRITON_PTR_TO_MEMREF_H
 #define TRITON_CONVERSION_TRITON_PTR_TO_MEMREF_TRITON_PTR_TO_MEMREF_H
 
diff --git a/include/triton-shared/Conversion/TritonToLinalgExperimental/CollapseShape.h b/include/triton-shared/Conversion/TritonToLinalgExperimental/CollapseShape.h
index 2c44951..064528e 100644
--- a/include/triton-shared/Conversion/TritonToLinalgExperimental/CollapseShape.h
+++ b/include/triton-shared/Conversion/TritonToLinalgExperimental/CollapseShape.h
@@ -1,6 +1,6 @@
 //===----------------------------------------------------------------------===//
 //
-// Copyright (c) Meta Platforms, Inc. and affiliates, Microsoft Corporation.
+// Copyright (c) Microsoft Corporation.
 // Licensed under the MIT license.
 //
 //===----------------------------------------------------------------------===//
diff --git a/include/triton-shared/Conversion/TritonToLinalgExperimental/Passes.h b/include/triton-shared/Conversion/TritonToLinalgExperimental/Passes.h
index fa5333f..5aa650a 100644
--- a/include/triton-shared/Conversion/TritonToLinalgExperimental/Passes.h
+++ b/include/triton-shared/Conversion/TritonToLinalgExperimental/Passes.h
@@ -1,6 +1,6 @@
 //===----------------------------------------------------------------------===//
 //
-// Copyright (c) Meta Platforms, Inc. and affiliates, Microsoft Corporation.
+// Copyright (c) Microsoft Corporation.
 // Licensed under the MIT license.
 //
 //===----------------------------------------------------------------------===//
diff --git a/include/triton-shared/Conversion/TritonToLinalgExperimental/Passes.td b/include/triton-shared/Conversion/TritonToLinalgExperimental/Passes.td
index 8ffae62..09d6655 100644
--- a/include/triton-shared/Conversion/TritonToLinalgExperimental/Passes.td
+++ b/include/triton-shared/Conversion/TritonToLinalgExperimental/Passes.td
@@ -1,6 +1,6 @@
 //===----------------------------------------------------------------------===//
 //
-// Copyright (c) Meta Platforms, Inc. and affiliates, Microsoft Corporation.
+// Copyright (c) Microsoft Corporation.
 // Licensed under the MIT license.
 //
 //===----------------------------------------------------------------------===//
diff --git a/include/triton-shared/Conversion/TritonToLinalgExperimental/ReconcilePtrCasts.h b/include/triton-shared/Conversion/TritonToLinalgExperimental/ReconcilePtrCasts.h
index 01f0229..bea24e8 100644
--- a/include/triton-shared/Conversion/TritonToLinalgExperimental/ReconcilePtrCasts.h
+++ b/include/triton-shared/Conversion/TritonToLinalgExperimental/ReconcilePtrCasts.h
@@ -1,6 +1,6 @@
 //===----------------------------------------------------------------------===//
 //
-// Copyright (c) Meta Platforms, Inc. and affiliates, Microsoft Corporation.
+// Copyright (c) Microsoft Corporation.
 // Licensed under the MIT license.
 //
 //===----------------------------------------------------------------------===//
diff --git a/include/triton-shared/Conversion/TritonToLinalgExperimental/TritonToLinalgExperimental.h b/include/triton-shared/Conversion/TritonToLinalgExperimental/TritonToLinalgExperimental.h
index f8a5f06..59b7b23 100644
--- a/include/triton-shared/Conversion/TritonToLinalgExperimental/TritonToLinalgExperimental.h
+++ b/include/triton-shared/Conversion/TritonToLinalgExperimental/TritonToLinalgExperimental.h
@@ -1,6 +1,6 @@
 //===----------------------------------------------------------------------===//
 //
-// Copyright (c) Meta Platforms, Inc. and affiliates, Microsoft Corporation.
+// Copyright (c) Microsoft Corporation.
 // Licensed under the MIT license.
 //
 //===----------------------------------------------------------------------===//
diff --git a/include/triton-shared/Conversion/TritonToLinalgExperimental/TritonToPtr.h b/include/triton-shared/Conversion/TritonToLinalgExperimental/TritonToPtr.h
index 8a3831b..305d5aa 100644
--- a/include/triton-shared/Conversion/TritonToLinalgExperimental/TritonToPtr.h
+++ b/include/triton-shared/Conversion/TritonToLinalgExperimental/TritonToPtr.h
@@ -1,6 +1,6 @@
 //===----------------------------------------------------------------------===//
 //
-// Copyright (c) Meta Platforms, Inc. and affiliates, Microsoft Corporation.
+// Copyright (c) Microsoft Corporation.
 // Licensed under the MIT license.
 //
 //===----------------------------------------------------------------------===//
diff --git a/include/triton-shared/Conversion/TritonToStructured/Passes.h b/include/triton-shared/Conversion/TritonToStructured/Passes.h
index 31bc380..3c3b81c 100644
--- a/include/triton-shared/Conversion/TritonToStructured/Passes.h
+++ b/include/triton-shared/Conversion/TritonToStructured/Passes.h
@@ -1,10 +1,3 @@
-//===----------------------------------------------------------------------===//
-//
-// Copyright (c) Meta Platforms, Inc. and affiliates, Microsoft Corporation.
-// Licensed under the MIT license.
-//
-//===----------------------------------------------------------------------===//
-
 #ifndef TRITON_TO_STRUCTURED_CONVERSION_PASSES_H
 #define TRITON_TO_STRUCTURED_CONVERSION_PASSES_H
 
diff --git a/include/triton-shared/Conversion/TritonToStructured/Passes.td b/include/triton-shared/Conversion/TritonToStructured/Passes.td
index 07ad788..ef61f0f 100644
--- a/include/triton-shared/Conversion/TritonToStructured/Passes.td
+++ b/include/triton-shared/Conversion/TritonToStructured/Passes.td
@@ -1,10 +1,3 @@
-//===----------------------------------------------------------------------===//
-//
-// Copyright (c) Meta Platforms, Inc. and affiliates, Microsoft Corporation.
-// Licensed under the MIT license.
-//
-//===----------------------------------------------------------------------===//
-
 #ifndef TRITON_TO_STRUCTURED_CONVERSION_PASSES
 #define TRITON_TO_STRUCTURED_CONVERSION_PASSES
 
diff --git a/include/triton-shared/Conversion/TritonToStructured/TritonToStructured.h b/include/triton-shared/Conversion/TritonToStructured/TritonToStructured.h
index d919cbb..bbf730e 100644
--- a/include/triton-shared/Conversion/TritonToStructured/TritonToStructured.h
+++ b/include/triton-shared/Conversion/TritonToStructured/TritonToStructured.h
@@ -1,10 +1,3 @@
-//===----------------------------------------------------------------------===//
-//
-// Copyright (c) Meta Platforms, Inc. and affiliates, Microsoft Corporation.
-// Licensed under the MIT license.
-//
-//===----------------------------------------------------------------------===//
-
 #ifndef TRITON_CONVERSION_TRITONTOSTRUCTURED_TRITONTOSTRUCTURED_H
 #define TRITON_CONVERSION_TRITONTOSTRUCTURED_TRITONTOSTRUCTURED_H
 
diff --git a/include/triton-shared/Conversion/TritonToUnstructured/Passes.h b/include/triton-shared/Conversion/TritonToUnstructured/Passes.h
index d0380b9..a2016c7 100644
--- a/include/triton-shared/Conversion/TritonToUnstructured/Passes.h
+++ b/include/triton-shared/Conversion/TritonToUnstructured/Passes.h
@@ -1,10 +1,3 @@
-//===----------------------------------------------------------------------===//
-//
-// Copyright (c) Meta Platforms, Inc. and affiliates, Microsoft Corporation.
-// Licensed under the MIT license.
-//
-//===----------------------------------------------------------------------===//
-
 #ifndef TRITON_TO_UNSTRUCTURED_CONVERSION_PASSES_H
 #define TRITON_TO_UNSTRUCTURED_CONVERSION_PASSES_H
 
diff --git a/include/triton-shared/Conversion/TritonToUnstructured/Passes.td b/include/triton-shared/Conversion/TritonToUnstructured/Passes.td
index cd7397c..542d087 100644
--- a/include/triton-shared/Conversion/TritonToUnstructured/Passes.td
+++ b/include/triton-shared/Conversion/TritonToUnstructured/Passes.td
@@ -1,10 +1,3 @@
-//===----------------------------------------------------------------------===//
-//
-// Copyright (c) Meta Platforms, Inc. and affiliates, Microsoft Corporation.
-// Licensed under the MIT license.
-//
-//===----------------------------------------------------------------------===//
-
 #ifndef TRITON_TO_UNSTRUCTURED_CONVERSION_PASSES
 #define TRITON_TO_UNSTRUCTURED_CONVERSION_PASSES
 
diff --git a/include/triton-shared/Conversion/TritonToUnstructured/TritonToUnstructured.h b/include/triton-shared/Conversion/TritonToUnstructured/TritonToUnstructured.h
index a36a343..03ccdcd 100644
--- a/include/triton-shared/Conversion/TritonToUnstructured/TritonToUnstructured.h
+++ b/include/triton-shared/Conversion/TritonToUnstructured/TritonToUnstructured.h
@@ -1,10 +1,3 @@
-//===----------------------------------------------------------------------===//
-//
-// Copyright (c) Meta Platforms, Inc. and affiliates, Microsoft Corporation.
-// Licensed under the MIT license.
-//
-//===----------------------------------------------------------------------===//
-
 #ifndef TRITON_CONVERSION_TRITON_TO_UNSTRUCTURED_TRITON_TO_UNSTRUCTURED_H
 #define TRITON_CONVERSION_TRITON_TO_UNSTRUCTURED_TRITON_TO_UNSTRUCTURED_H
 
diff --git a/include/triton-shared/Conversion/UnstructuredToMemref/Passes.h b/include/triton-shared/Conversion/UnstructuredToMemref/Passes.h
index db664ef..f2d7117 100644
--- a/include/triton-shared/Conversion/UnstructuredToMemref/Passes.h
+++ b/include/triton-shared/Conversion/UnstructuredToMemref/Passes.h
@@ -1,6 +1,6 @@
 //===----------------------------------------------------------------------===//
 //
-// Copyright (c) Meta Platforms, Inc. and affiliates, Microsoft Corporation.
+// Copyright (c) Microsoft Corporation.
 // Licensed under the MIT license.
 //
 //===----------------------------------------------------------------------===//
diff --git a/include/triton-shared/Conversion/UnstructuredToMemref/Passes.td b/include/triton-shared/Conversion/UnstructuredToMemref/Passes.td
index 4a7b382..a0bf316 100644
--- a/include/triton-shared/Conversion/UnstructuredToMemref/Passes.td
+++ b/include/triton-shared/Conversion/UnstructuredToMemref/Passes.td
@@ -1,6 +1,6 @@
 //===----------------------------------------------------------------------===//
 //
-// Copyright (c) Meta Platforms, Inc. and affiliates, Microsoft Corporation.
+// Copyright (c) Microsoft Corporation.
 // Licensed under the MIT license.
 //
 //===----------------------------------------------------------------------===//
diff --git a/include/triton-shared/Conversion/UnstructuredToMemref/UnstructuredToMemref.h b/include/triton-shared/Conversion/UnstructuredToMemref/UnstructuredToMemref.h
index 4ad6d0a..ad0f5c4 100644
--- a/include/triton-shared/Conversion/UnstructuredToMemref/UnstructuredToMemref.h
+++ b/include/triton-shared/Conversion/UnstructuredToMemref/UnstructuredToMemref.h
@@ -1,6 +1,6 @@
 //===----------------------------------------------------------------------===//
 //
-// Copyright (c) Meta Platforms, Inc. and affiliates, Microsoft Corporation.
+// Copyright (c) Microsoft Corporation.
 // Licensed under the MIT license.
 //
 //===----------------------------------------------------------------------===//
diff --git a/include/triton-shared/Dialect/TPtr/IR/TPtrDialect.h b/include/triton-shared/Dialect/TPtr/IR/TPtrDialect.h
index f810b40..c7e5933 100644
--- a/include/triton-shared/Dialect/TPtr/IR/TPtrDialect.h
+++ b/include/triton-shared/Dialect/TPtr/IR/TPtrDialect.h
@@ -1,10 +1,3 @@
-//===----------------------------------------------------------------------===//
-//
-// Copyright (c) Meta Platforms, Inc. and affiliates, Microsoft Corporation.
-// Licensed under the MIT license.
-//
-//===----------------------------------------------------------------------===//
-
 #ifndef MLIR_DIALECT_TPTR_IR_TPTR_DIALECT_H_
 #define MLIR_DIALECT_TPTR_IR_TPTR_DIALECT_H_
 
diff --git a/include/triton-shared/Dialect/TPtr/IR/TPtrDialect.td b/include/triton-shared/Dialect/TPtr/IR/TPtrDialect.td
index f724f78..04d894f 100644
--- a/include/triton-shared/Dialect/TPtr/IR/TPtrDialect.td
+++ b/include/triton-shared/Dialect/TPtr/IR/TPtrDialect.td
@@ -1,10 +1,3 @@
-//===----------------------------------------------------------------------===//
-//
-// Copyright (c) Meta Platforms, Inc. and affiliates, Microsoft Corporation.
-// Licensed under the MIT license.
-//
-//===----------------------------------------------------------------------===//
-
 #ifndef TPTR_DIALECT
 #define TPTR_DIALECT
 
diff --git a/include/triton-shared/Dialect/TritonStructured/IR/TritonStructuredDialect.h b/include/triton-shared/Dialect/TritonStructured/IR/TritonStructuredDialect.h
index af72fbe..fbead0c 100644
--- a/include/triton-shared/Dialect/TritonStructured/IR/TritonStructuredDialect.h
+++ b/include/triton-shared/Dialect/TritonStructured/IR/TritonStructuredDialect.h
@@ -1,10 +1,3 @@
-//===----------------------------------------------------------------------===//
-//
-// Copyright (c) Meta Platforms, Inc. and affiliates, Microsoft Corporation.
-// Licensed under the MIT license.
-//
-//===----------------------------------------------------------------------===//
-
 #ifndef MLIR_DIALECT_TRITON_STRUCTURED_IR_TRITON_STRUCTURED_DIALECT_H_
 #define MLIR_DIALECT_TRITON_STRUCTURED_IR_TRITON_STRUCTURED_DIALECT_H_
 
diff --git a/include/triton-shared/Dialect/TritonStructured/IR/TritonStructuredDialect.td b/include/triton-shared/Dialect/TritonStructured/IR/TritonStructuredDialect.td
index e5c81b2..17c3ce9 100644
--- a/include/triton-shared/Dialect/TritonStructured/IR/TritonStructuredDialect.td
+++ b/include/triton-shared/Dialect/TritonStructured/IR/TritonStructuredDialect.td
@@ -1,10 +1,3 @@
-//===----------------------------------------------------------------------===//
-//
-// Copyright (c) Meta Platforms, Inc. and affiliates, Microsoft Corporation.
-// Licensed under the MIT license.
-//
-//===----------------------------------------------------------------------===//
-
 #ifndef TRITON_STRUCTURED_DIALECT
 #define TRITON_STRUCTURED_DIALECT
 
diff --git a/include/triton-shared/Dialect/TritonTilingExt/IR/TritonTilingExtDialect.h b/include/triton-shared/Dialect/TritonTilingExt/IR/TritonTilingExtDialect.h
index 5cf8492..53e031d 100644
--- a/include/triton-shared/Dialect/TritonTilingExt/IR/TritonTilingExtDialect.h
+++ b/include/triton-shared/Dialect/TritonTilingExt/IR/TritonTilingExtDialect.h
@@ -1,6 +1,6 @@
 //===----------------------------------------------------------------------===//
 //
-// Copyright (c) Meta Platforms, Inc. and affiliates, Microsoft Corporation.
+// Copyright (c) Microsoft Corporation.
 // Licensed under the MIT license.
 //
 //===----------------------------------------------------------------------===//
diff --git a/include/triton-shared/Dialect/TritonTilingExt/IR/TritonTilingExtInterfaces.td b/include/triton-shared/Dialect/TritonTilingExt/IR/TritonTilingExtInterfaces.td
index 567d95e..e74fbb6 100644
--- a/include/triton-shared/Dialect/TritonTilingExt/IR/TritonTilingExtInterfaces.td
+++ b/include/triton-shared/Dialect/TritonTilingExt/IR/TritonTilingExtInterfaces.td
@@ -1,6 +1,6 @@
 //===----------------------------------------------------------------------===//
 //
-// Copyright (c) Meta Platforms, Inc. and affiliates, Microsoft Corporation.
+// Copyright (c) Microsoft Corporation.
 // Licensed under the MIT license.
 //
 //===----------------------------------------------------------------------===//
diff --git a/include/triton-shared/Dialect/TritonTilingExt/IR/TritonTilingExtOps.td b/include/triton-shared/Dialect/TritonTilingExt/IR/TritonTilingExtOps.td
index d589aab..d3a4268 100644
--- a/include/triton-shared/Dialect/TritonTilingExt/IR/TritonTilingExtOps.td
+++ b/include/triton-shared/Dialect/TritonTilingExt/IR/TritonTilingExtOps.td
@@ -1,6 +1,6 @@
 //===----------------------------------------------------------------------===//
 //
-// Copyright (c) Meta Platforms, Inc. and affiliates, Microsoft Corporation.
+// Copyright (c) Microsoft Corporation.
 // Licensed under the MIT license.
 //
 //===----------------------------------------------------------------------===//
diff --git a/include/triton-shared/Transform/AddLLVMDebugInfo/AddLLVMDebugInfo.h b/include/triton-shared/Transform/AddLLVMDebugInfo/AddLLVMDebugInfo.h
index d2fc8f4..e350e8a 100644
--- a/include/triton-shared/Transform/AddLLVMDebugInfo/AddLLVMDebugInfo.h
+++ b/include/triton-shared/Transform/AddLLVMDebugInfo/AddLLVMDebugInfo.h
@@ -1,6 +1,6 @@
 //===----------------------------------------------------------------------===//
 //
-// Copyright (c) Meta Platforms, Inc. and affiliates, Microsoft Corporation.
+// Copyright (c) Microsoft Corporation.
 // Licensed under the MIT license.
 //
 //===----------------------------------------------------------------------===//
diff --git a/include/triton-shared/Transform/AddLLVMDebugInfo/Passes.h b/include/triton-shared/Transform/AddLLVMDebugInfo/Passes.h
index 69b127e..9c94ac4 100644
--- a/include/triton-shared/Transform/AddLLVMDebugInfo/Passes.h
+++ b/include/triton-shared/Transform/AddLLVMDebugInfo/Passes.h
@@ -1,6 +1,6 @@
 //===----------------------------------------------------------------------===//
 //
-// Copyright (c) Meta Platforms, Inc. and affiliates, Microsoft Corporation.
+// Copyright (c) Microsoft Corporation.
 // Licensed under the MIT license.
 //
 //===----------------------------------------------------------------------===//
diff --git a/include/triton-shared/Transform/AddLLVMDebugInfo/Passes.td b/include/triton-shared/Transform/AddLLVMDebugInfo/Passes.td
index 1d606d7..e391eb2 100644
--- a/include/triton-shared/Transform/AddLLVMDebugInfo/Passes.td
+++ b/include/triton-shared/Transform/AddLLVMDebugInfo/Passes.td
@@ -1,6 +1,6 @@
 //===----------------------------------------------------------------------===//
 //
-// Copyright (c) Meta Platforms, Inc. and affiliates, Microsoft Corporation.
+// Copyright (c) Microsoft Corporation.
 // Licensed under the MIT license.
 //
 //===----------------------------------------------------------------------===//
diff --git a/include/triton-shared/Utils/Utils.h b/include/triton-shared/Utils/Utils.h
index dfc693a..e484bee 100644
--- a/include/triton-shared/Utils/Utils.h
+++ b/include/triton-shared/Utils/Utils.h
@@ -1,10 +1,3 @@
-//===----------------------------------------------------------------------===//
-//
-// Copyright (c) Meta Platforms, Inc. and affiliates, Microsoft Corporation.
-// Licensed under the MIT license.
-//
-//===----------------------------------------------------------------------===//
-
 #ifndef TRITON_SHARED_UTILITY_H
 #define TRITON_SHARED_UTILITY_H
 
diff --git a/lib/Analysis/MaskAnalysis.cpp b/lib/Analysis/MaskAnalysis.cpp
index 9ca1a92..5b84859 100644
--- a/lib/Analysis/MaskAnalysis.cpp
+++ b/lib/Analysis/MaskAnalysis.cpp
@@ -1,6 +1,6 @@
 //===----------------------------------------------------------------------===//
 //
-// Copyright (c) Meta Platforms, Inc. and affiliates, Microsoft Corporation.
+// Copyright (c) Microsoft Corporation.
 // Licensed under the MIT license.
 //
 //===----------------------------------------------------------------------===//
diff --git a/lib/Analysis/OpFoldResultUtils.cpp b/lib/Analysis/OpFoldResultUtils.cpp
index d0b58d4..85d7842 100644
--- a/lib/Analysis/OpFoldResultUtils.cpp
+++ b/lib/Analysis/OpFoldResultUtils.cpp
@@ -1,6 +1,6 @@
 //===----------------------------------------------------------------------===//
 //
-// Copyright (c) Meta Platforms, Inc. and affiliates, Microsoft Corporation.
+// Copyright (c) Microsoft Corporation, Meta Platforms.
 // Licensed under the MIT license.
 //
 //===----------------------------------------------------------------------===//
diff --git a/lib/Analysis/PtrAnalysis.cpp b/lib/Analysis/PtrAnalysis.cpp
index 0fe5f84..1a17567 100644
--- a/lib/Analysis/PtrAnalysis.cpp
+++ b/lib/Analysis/PtrAnalysis.cpp
@@ -1,6 +1,6 @@
 //===----------------------------------------------------------------------===//
 //
-// Copyright (c) Meta Platforms, Inc. and affiliates, Microsoft Corporation.
+// Copyright (c) Microsoft Corporation.
 // Licensed under the MIT license.
 //
 //===----------------------------------------------------------------------===//
diff --git a/lib/Analysis/UseAnalysis.cpp b/lib/Analysis/UseAnalysis.cpp
index 7bde980..62e4508 100644
--- a/lib/Analysis/UseAnalysis.cpp
+++ b/lib/Analysis/UseAnalysis.cpp
@@ -1,6 +1,6 @@
 //===----------------------------------------------------------------------===//
 //
-// Copyright (c) Meta Platforms, Inc. and affiliates, Microsoft Corporation.
+// Copyright (c) Microsoft Corporation.
 // Licensed under the MIT license.
 //
 //===----------------------------------------------------------------------===//
diff --git a/lib/AnalysisStructured/PtrAnalysis.cpp b/lib/AnalysisStructured/PtrAnalysis.cpp
index 0391a15..e5b6ece 100644
--- a/lib/AnalysisStructured/PtrAnalysis.cpp
+++ b/lib/AnalysisStructured/PtrAnalysis.cpp
@@ -1,6 +1,6 @@
 //===----------------------------------------------------------------------===//
 //
-// Copyright (c) Meta Platforms, Inc. and affiliates, Microsoft Corporation.
+// Copyright (c) Microsoft Corporation, Meta Platforms.
 // Licensed under the MIT license.
 //
 //===----------------------------------------------------------------------===//
diff --git a/lib/Conversion/StructuredToMemref/StructuredToMemref.cpp b/lib/Conversion/StructuredToMemref/StructuredToMemref.cpp
index 46a56df..6b0ddb8 100644
--- a/lib/Conversion/StructuredToMemref/StructuredToMemref.cpp
+++ b/lib/Conversion/StructuredToMemref/StructuredToMemref.cpp
@@ -1,6 +1,6 @@
 //===----------------------------------------------------------------------===//
 //
-// Copyright (c) Meta Platforms, Inc. and affiliates, Microsoft Corporation.
+// Copyright (c) Microsoft Corporation, Meta Platforms.
 // Licensed under the MIT license.
 //
 //===----------------------------------------------------------------------===//
diff --git a/lib/Conversion/StructuredToMemref/StructuredToMemrefPass.cpp b/lib/Conversion/StructuredToMemref/StructuredToMemrefPass.cpp
index 43a9210..2a393e8 100644
--- a/lib/Conversion/StructuredToMemref/StructuredToMemrefPass.cpp
+++ b/lib/Conversion/StructuredToMemref/StructuredToMemrefPass.cpp
@@ -1,6 +1,6 @@
 //===----------------------------------------------------------------------===//
 //
-// Copyright (c) Meta Platforms, Inc. and affiliates, Microsoft Corporation.
+// Copyright (c) Microsoft Corporation, Meta Platforms.
 // Licensed under the MIT license.
 //
 //===----------------------------------------------------------------------===//
diff --git a/lib/Conversion/TritonArithToLinalg/TritonArithToLinalg.cpp b/lib/Conversion/TritonArithToLinalg/TritonArithToLinalg.cpp
index 2359a27..92b6856 100644
--- a/lib/Conversion/TritonArithToLinalg/TritonArithToLinalg.cpp
+++ b/lib/Conversion/TritonArithToLinalg/TritonArithToLinalg.cpp
@@ -1,6 +1,6 @@
 //===----------------------------------------------------------------------===//
 //
-// Copyright (c) Meta Platforms, Inc. and affiliates, Microsoft Corporation.
+// Copyright (c) Microsoft Corporation, Meta Platforms.
 // Licensed under the MIT license.
 //
 //===----------------------------------------------------------------------===//
diff --git a/lib/Conversion/TritonArithToLinalg/TritonArithToLinalgPass.cpp b/lib/Conversion/TritonArithToLinalg/TritonArithToLinalgPass.cpp
index c7186df..39cf0ba 100644
--- a/lib/Conversion/TritonArithToLinalg/TritonArithToLinalgPass.cpp
+++ b/lib/Conversion/TritonArithToLinalg/TritonArithToLinalgPass.cpp
@@ -1,6 +1,6 @@
 //===----------------------------------------------------------------------===//
 //
-// Copyright (c) Meta Platforms, Inc. and affiliates, Microsoft Corporation.
+// Copyright (c) Microsoft Corporation, Meta Platforms.
 // Licensed under the MIT license.
 //
 //===----------------------------------------------------------------------===//
diff --git a/lib/Conversion/TritonPtrToMemref/TritonPtrToMemrefPass.cpp b/lib/Conversion/TritonPtrToMemref/TritonPtrToMemrefPass.cpp
index cd3ca0d..d462f4d 100644
--- a/lib/Conversion/TritonPtrToMemref/TritonPtrToMemrefPass.cpp
+++ b/lib/Conversion/TritonPtrToMemref/TritonPtrToMemrefPass.cpp
@@ -1,6 +1,6 @@
 //===----------------------------------------------------------------------===//
 //
-// Copyright (c) Meta Platforms, Inc. and affiliates, Microsoft Corporation.
+// Copyright (c) Microsoft Corporation.
 // Licensed under the MIT license.
 //
 //===----------------------------------------------------------------------===//
diff --git a/lib/Conversion/TritonToLinalgExperimental/CollapseShape.cpp b/lib/Conversion/TritonToLinalgExperimental/CollapseShape.cpp
index 063efb0..1adf485 100644
--- a/lib/Conversion/TritonToLinalgExperimental/CollapseShape.cpp
+++ b/lib/Conversion/TritonToLinalgExperimental/CollapseShape.cpp
@@ -1,6 +1,6 @@
 //===----------------------------------------------------------------------===//
 //
-// Copyright (c) Meta Platforms, Inc. and affiliates, Microsoft Corporation.
+// Copyright (c) Microsoft Corporation.
 // Licensed under the MIT license.
 //
 //===----------------------------------------------------------------------===//
diff --git a/lib/Conversion/TritonToLinalgExperimental/ReconcilePtrCastsPass.cpp b/lib/Conversion/TritonToLinalgExperimental/ReconcilePtrCastsPass.cpp
index b128012..a25a713 100644
--- a/lib/Conversion/TritonToLinalgExperimental/ReconcilePtrCastsPass.cpp
+++ b/lib/Conversion/TritonToLinalgExperimental/ReconcilePtrCastsPass.cpp
@@ -1,6 +1,6 @@
 //===----------------------------------------------------------------------===//
 //
-// Copyright (c) Meta Platforms, Inc. and affiliates, Microsoft Corporation.
+// Copyright (c) Microsoft Corporation.
 // Licensed under the MIT license.
 //
 //===----------------------------------------------------------------------===//
diff --git a/lib/Conversion/TritonToLinalgExperimental/TritonToLinalgExperimentalPass.cpp b/lib/Conversion/TritonToLinalgExperimental/TritonToLinalgExperimentalPass.cpp
index 58e94fe..e0d968d 100644
--- a/lib/Conversion/TritonToLinalgExperimental/TritonToLinalgExperimentalPass.cpp
+++ b/lib/Conversion/TritonToLinalgExperimental/TritonToLinalgExperimentalPass.cpp
@@ -1,6 +1,6 @@
 //===----------------------------------------------------------------------===//
 //
-// Copyright (c) Meta Platforms, Inc. and affiliates, Microsoft Corporation.
+// Copyright (c) Microsoft Corporation.
 // Licensed under the MIT license.
 //
 //===----------------------------------------------------------------------===//
diff --git a/lib/Conversion/TritonToLinalgExperimental/TritonToPtrPass.cpp b/lib/Conversion/TritonToLinalgExperimental/TritonToPtrPass.cpp
index d78fded..6e3cc9d 100644
--- a/lib/Conversion/TritonToLinalgExperimental/TritonToPtrPass.cpp
+++ b/lib/Conversion/TritonToLinalgExperimental/TritonToPtrPass.cpp
@@ -1,6 +1,6 @@
 //===----------------------------------------------------------------------===//
 //
-// Copyright (c) Meta Platforms, Inc. and affiliates, Microsoft Corporation.
+// Copyright (c) Microsoft Corporation.
 // Licensed under the MIT license.
 //
 //===----------------------------------------------------------------------===//
diff --git a/lib/Conversion/TritonToStructured/TritonToStructuredPass.cpp b/lib/Conversion/TritonToStructured/TritonToStructuredPass.cpp
index cbfebe1..f75f347 100644
--- a/lib/Conversion/TritonToStructured/TritonToStructuredPass.cpp
+++ b/lib/Conversion/TritonToStructured/TritonToStructuredPass.cpp
@@ -1,6 +1,6 @@
 //===----------------------------------------------------------------------===//
 //
-// Copyright (c) Meta Platforms, Inc. and affiliates, Microsoft Corporation.
+// Copyright (c) Microsoft Corporation, Meta Platforms.
 // Licensed under the MIT license.
 //
 //===----------------------------------------------------------------------===//
diff --git a/lib/Conversion/TritonToUnstructured/TritonToUnstructuredPass.cpp b/lib/Conversion/TritonToUnstructured/TritonToUnstructuredPass.cpp
index f453be6..45cb8be 100644
--- a/lib/Conversion/TritonToUnstructured/TritonToUnstructuredPass.cpp
+++ b/lib/Conversion/TritonToUnstructured/TritonToUnstructuredPass.cpp
@@ -1,6 +1,6 @@
 //===----------------------------------------------------------------------===//
 //
-// Copyright (c) Meta Platforms, Inc. and affiliates, Microsoft Corporation.
+// Copyright (c) Microsoft Corporation.
 // Licensed under the MIT license.
 //
 //===----------------------------------------------------------------------===//
diff --git a/lib/Conversion/UnstructuredToMemref/UnstructuredToMemrefPass.cpp b/lib/Conversion/UnstructuredToMemref/UnstructuredToMemrefPass.cpp
index 36714c1..076e4eb 100644
--- a/lib/Conversion/UnstructuredToMemref/UnstructuredToMemrefPass.cpp
+++ b/lib/Conversion/UnstructuredToMemref/UnstructuredToMemrefPass.cpp
@@ -1,6 +1,6 @@
 //===----------------------------------------------------------------------===//
 //
-// Copyright (c) Meta Platforms, Inc. and affiliates, Microsoft Corporation.
+// Copyright (c) Microsoft Corporation.
 // Licensed under the MIT license.
 //
 //===----------------------------------------------------------------------===//
diff --git a/lib/Dialect/TPtr/IR/TPtrDialect.cpp b/lib/Dialect/TPtr/IR/TPtrDialect.cpp
index 5ec3976..c4b8b7c 100644
--- a/lib/Dialect/TPtr/IR/TPtrDialect.cpp
+++ b/lib/Dialect/TPtr/IR/TPtrDialect.cpp
@@ -1,10 +1,3 @@
-//===----------------------------------------------------------------------===//
-//
-// Copyright (c) Meta Platforms, Inc. and affiliates, Microsoft Corporation.
-// Licensed under the MIT license.
-//
-//===----------------------------------------------------------------------===//
-
 #include "mlir/IR/Builders.h"
 
 #include "triton-shared/Dialect/TPtr/IR/TPtrDialect.h"
diff --git a/lib/Dialect/TPtr/IR/TPtrOps.cpp b/lib/Dialect/TPtr/IR/TPtrOps.cpp
index 06bffd5..dad589e 100644
--- a/lib/Dialect/TPtr/IR/TPtrOps.cpp
+++ b/lib/Dialect/TPtr/IR/TPtrOps.cpp
@@ -1,10 +1,3 @@
-//===----------------------------------------------------------------------===//
-//
-// Copyright (c) Meta Platforms, Inc. and affiliates, Microsoft Corporation.
-// Licensed under the MIT license.
-//
-//===----------------------------------------------------------------------===//
-
 #include "mlir/Bytecode/BytecodeOpInterface.h"
 #include "mlir/Interfaces/SideEffectInterfaces.h" // Required for IR/TPtrOps.h.inc
 
diff --git a/lib/Dialect/TritonStructured/IR/TritonStructuredDialect.cpp b/lib/Dialect/TritonStructured/IR/TritonStructuredDialect.cpp
index 0a9ce77..2af19b8 100644
--- a/lib/Dialect/TritonStructured/IR/TritonStructuredDialect.cpp
+++ b/lib/Dialect/TritonStructured/IR/TritonStructuredDialect.cpp
@@ -1,10 +1,3 @@
-//===----------------------------------------------------------------------===//
-//
-// Copyright (c) Meta Platforms, Inc. and affiliates, Microsoft Corporation.
-// Licensed under the MIT license.
-//
-//===----------------------------------------------------------------------===//
-
 #include "triton-shared/Dialect/TritonStructured/IR/TritonStructuredDialect.h"
 
 using namespace mlir;
diff --git a/lib/Dialect/TritonStructured/IR/TritonStructuredOps.cpp b/lib/Dialect/TritonStructured/IR/TritonStructuredOps.cpp
index dbbb0e4..150e926 100644
--- a/lib/Dialect/TritonStructured/IR/TritonStructuredOps.cpp
+++ b/lib/Dialect/TritonStructured/IR/TritonStructuredOps.cpp
@@ -1,10 +1,3 @@
-//===----------------------------------------------------------------------===//
-//
-// Copyright (c) Meta Platforms, Inc. and affiliates, Microsoft Corporation.
-// Licensed under the MIT license.
-//
-//===----------------------------------------------------------------------===//
-
 #include "triton/Dialect/Triton/IR/Dialect.h"
 #include "triton/Dialect/Triton/IR/Types.h"
 
diff --git a/lib/Dialect/TritonTilingExt/IR/BufferizableOpInterfaceImpl.cpp b/lib/Dialect/TritonTilingExt/IR/BufferizableOpInterfaceImpl.cpp
index d992f2a..d98d763 100644
--- a/lib/Dialect/TritonTilingExt/IR/BufferizableOpInterfaceImpl.cpp
+++ b/lib/Dialect/TritonTilingExt/IR/BufferizableOpInterfaceImpl.cpp
@@ -1,6 +1,6 @@
 //===----------------------------------------------------------------------===//
 //
-// Copyright (c) Meta Platforms, Inc. and affiliates, Microsoft Corporation.
+// Copyright (c) Microsoft Corporation.
 // Licensed under the MIT license.
 //
 //===----------------------------------------------------------------------===//
diff --git a/lib/Dialect/TritonTilingExt/IR/CumSum.cpp b/lib/Dialect/TritonTilingExt/IR/CumSum.cpp
index ebada13..619b653 100644
--- a/lib/Dialect/TritonTilingExt/IR/CumSum.cpp
+++ b/lib/Dialect/TritonTilingExt/IR/CumSum.cpp
@@ -1,6 +1,6 @@
 //===----------------------------------------------------------------------===//
 //
-// Copyright (c) Meta Platforms, Inc. and affiliates, Microsoft Corporation.
+// Copyright (c) Microsoft Corporation.
 // Licensed under the MIT license.
 //
 //===----------------------------------------------------------------------===//
diff --git a/lib/Dialect/TritonTilingExt/IR/TritonTilingExtDialect.cpp b/lib/Dialect/TritonTilingExt/IR/TritonTilingExtDialect.cpp
index 598ae39..acf118a 100644
--- a/lib/Dialect/TritonTilingExt/IR/TritonTilingExtDialect.cpp
+++ b/lib/Dialect/TritonTilingExt/IR/TritonTilingExtDialect.cpp
@@ -1,6 +1,6 @@
 //===----------------------------------------------------------------------===//
 //
-// Copyright (c) Meta Platforms, Inc. and affiliates, Microsoft Corporation.
+// Copyright (c) Microsoft Corporation.
 // Licensed under the MIT license.
 //
 //===----------------------------------------------------------------------===//
diff --git a/lib/Sanitizer/SanitizerAttributes/SanitizerAttributes.cpp b/lib/Sanitizer/SanitizerAttributes/SanitizerAttributes.cpp
index 32e508a..2733a55 100644
--- a/lib/Sanitizer/SanitizerAttributes/SanitizerAttributes.cpp
+++ b/lib/Sanitizer/SanitizerAttributes/SanitizerAttributes.cpp
@@ -1,10 +1,3 @@
-//===----------------------------------------------------------------------===//
-//
-// Copyright (c) Meta Platforms, Inc. and affiliates, Microsoft Corporation.
-// Licensed under the MIT license.
-//
-//===----------------------------------------------------------------------===//
-
 #include "llvm/Passes/PassBuilder.h"
 #include "llvm/Plugins/PassPlugin.h"
 #include "llvm/Support/CommandLine.h"
diff --git a/lib/Transform/AddLLVMDebugInfo/AddLLVMDebugInfoPass.cpp b/lib/Transform/AddLLVMDebugInfo/AddLLVMDebugInfoPass.cpp
index f2f505e..a72431a 100644
--- a/lib/Transform/AddLLVMDebugInfo/AddLLVMDebugInfoPass.cpp
+++ b/lib/Transform/AddLLVMDebugInfo/AddLLVMDebugInfoPass.cpp
@@ -1,6 +1,6 @@
 //===----------------------------------------------------------------------===//
 //
-// Copyright (c) Meta Platforms, Inc. and affiliates, Microsoft Corporation.
+// Copyright (c) Microsoft Corporation.
 // Licensed under the MIT license.
 //
 //===----------------------------------------------------------------------===//
diff --git a/lib/Utils/Utils.cpp b/lib/Utils/Utils.cpp
index 7abd47b..2483d53 100644
--- a/lib/Utils/Utils.cpp
+++ b/lib/Utils/Utils.cpp
@@ -1,10 +1,3 @@
-//===----------------------------------------------------------------------===//
-//
-// Copyright (c) Meta Platforms, Inc. and affiliates, Microsoft Corporation.
-// Licensed under the MIT license.
-//
-//===----------------------------------------------------------------------===//
-
 #include "triton/Dialect/Triton/IR/Dialect.h"
 
 namespace mlir {
diff --git a/python/examples/bare_matmul.py b/python/examples/bare_matmul.py
index 1793c31..eaa408b 100644
--- a/python/examples/bare_matmul.py
+++ b/python/examples/bare_matmul.py
@@ -1,6 +1,3 @@
-# Copyright (c) Meta Platforms, Inc. and affiliates, Microsoft Corporation.
-# Licensed under the MIT license.
-
 # this is a benchmark which multiplies square matrices with maximum block size
 # to check the performance of tl.dot operation
 
diff --git a/python/examples/benchmark.py b/python/examples/benchmark.py
index a95c326..f82d344 100644
--- a/python/examples/benchmark.py
+++ b/python/examples/benchmark.py
@@ -1,6 +1,3 @@
-# Copyright (c) Meta Platforms, Inc. and affiliates, Microsoft Corporation.
-# Licensed under the MIT license.
-
 import time
 import numpy as np
 from functools import wraps
diff --git a/python/examples/conftest.py b/python/examples/conftest.py
index 5b00194..33397c5 100644
--- a/python/examples/conftest.py
+++ b/python/examples/conftest.py
@@ -1,6 +1,3 @@
-# Copyright (c) Meta Platforms, Inc. and affiliates, Microsoft Corporation.
-# Licensed under the MIT license.
-
 import pytest
 import os
 import tempfile
diff --git a/python/examples/test_addptr.py b/python/examples/test_addptr.py
index 53e4ffc..5239c85 100644
--- a/python/examples/test_addptr.py
+++ b/python/examples/test_addptr.py
@@ -1,6 +1,3 @@
-# Copyright (c) Meta Platforms, Inc. and affiliates, Microsoft Corporation.
-# Licensed under the MIT license.
-
 import torch
 
 import triton
diff --git a/python/examples/test_blockptr_complex_offset.py b/python/examples/test_blockptr_complex_offset.py
index 2123f1f..94c0cd5 100644
--- a/python/examples/test_blockptr_complex_offset.py
+++ b/python/examples/test_blockptr_complex_offset.py
@@ -1,6 +1,3 @@
-# Copyright (c) Meta Platforms, Inc. and affiliates, Microsoft Corporation.
-# Licensed under the MIT license.
-
 import torch
 
 import triton
diff --git a/python/examples/test_early_return.py b/python/examples/test_early_return.py
index 231ea78..2bb59be 100644
--- a/python/examples/test_early_return.py
+++ b/python/examples/test_early_return.py
@@ -1,6 +1,3 @@
-# Copyright (c) Meta Platforms, Inc. and affiliates, Microsoft Corporation.
-# Licensed under the MIT license.
-
 import torch
 
 import triton
diff --git a/python/examples/test_einsum_qhmd_hmpd_to_qhmp.py b/python/examples/test_einsum_qhmd_hmpd_to_qhmp.py
index a07553c..ffaa84e 100644
--- a/python/examples/test_einsum_qhmd_hmpd_to_qhmp.py
+++ b/python/examples/test_einsum_qhmd_hmpd_to_qhmp.py
@@ -1,6 +1,3 @@
-# Copyright (c) Meta Platforms, Inc. and affiliates, Microsoft Corporation.
-# Licensed under the MIT license.
-
 import torch
 
 import triton
diff --git a/python/examples/test_gather_scatter.py b/python/examples/test_gather_scatter.py
index e448063..db08736 100644
--- a/python/examples/test_gather_scatter.py
+++ b/python/examples/test_gather_scatter.py
@@ -1,6 +1,3 @@
-# Copyright (c) Meta Platforms, Inc. and affiliates, Microsoft Corporation.
-# Licensed under the MIT license.
-
 import torch
 
 import triton
diff --git a/python/examples/test_gather_scatter_continuous.py b/python/examples/test_gather_scatter_continuous.py
index 4126408..98fe5c0 100644
--- a/python/examples/test_gather_scatter_continuous.py
+++ b/python/examples/test_gather_scatter_continuous.py
@@ -1,6 +1,3 @@
-# Copyright (c) Meta Platforms, Inc. and affiliates, Microsoft Corporation.
-# Licensed under the MIT license.
-
 import pytest
 import torch
 
diff --git a/python/examples/test_index_select.py b/python/examples/test_index_select.py
index 76de60a..c28b8b1 100644
--- a/python/examples/test_index_select.py
+++ b/python/examples/test_index_select.py
@@ -1,6 +1,3 @@
-# Copyright (c) Meta Platforms, Inc. and affiliates, Microsoft Corporation.
-# Licensed under the MIT license.
-
 import pytest
 import torch
 
diff --git a/python/examples/test_layernorm.py b/python/examples/test_layernorm.py
index abb4dd7..dc9b847 100644
--- a/python/examples/test_layernorm.py
+++ b/python/examples/test_layernorm.py
@@ -1,6 +1,3 @@
-# Copyright (c) Meta Platforms, Inc. and affiliates, Microsoft Corporation.
-# Licensed under the MIT license.
-
 # This is the Layer Norm forward pass from the Triton tutorial found here:
 # https://github.com/triton-lang/triton/blob/main/python/tutorials/05-layer-norm.py
 
diff --git a/python/examples/test_load_2d_tensor_block.py b/python/examples/test_load_2d_tensor_block.py
index 78d86a0..ec7d53d 100644
--- a/python/examples/test_load_2d_tensor_block.py
+++ b/python/examples/test_load_2d_tensor_block.py
@@ -1,6 +1,3 @@
-# Copyright (c) Meta Platforms, Inc. and affiliates, Microsoft Corporation.
-# Licensed under the MIT license.
-
 import torch
 
 import triton
diff --git a/python/examples/test_load_2d_tensor_col.py b/python/examples/test_load_2d_tensor_col.py
index 17055fd..f1ed8ad 100644
--- a/python/examples/test_load_2d_tensor_col.py
+++ b/python/examples/test_load_2d_tensor_col.py
@@ -1,6 +1,3 @@
-# Copyright (c) Meta Platforms, Inc. and affiliates, Microsoft Corporation.
-# Licensed under the MIT license.
-
 import torch
 
 import triton
diff --git a/python/examples/test_mask.py b/python/examples/test_mask.py
index 9253187..bc063c1 100644
--- a/python/examples/test_mask.py
+++ b/python/examples/test_mask.py
@@ -1,6 +1,3 @@
-# Copyright (c) Meta Platforms, Inc. and affiliates, Microsoft Corporation.
-# Licensed under the MIT license.
-
 import torch
 
 import triton
diff --git a/python/examples/test_mask_loop_iter_arg.py b/python/examples/test_mask_loop_iter_arg.py
index ffef270..3677ed7 100644
--- a/python/examples/test_mask_loop_iter_arg.py
+++ b/python/examples/test_mask_loop_iter_arg.py
@@ -1,6 +1,3 @@
-# Copyright (c) Meta Platforms, Inc. and affiliates, Microsoft Corporation.
-# Licensed under the MIT license.
-
 import torch
 import triton
 import pytest
diff --git a/python/examples/test_matmul.py b/python/examples/test_matmul.py
index 2ea8fe2..470006a 100644
--- a/python/examples/test_matmul.py
+++ b/python/examples/test_matmul.py
@@ -1,6 +1,3 @@
-# Copyright (c) Meta Platforms, Inc. and affiliates, Microsoft Corporation.
-# Licensed under the MIT license.
-
 import torch
 
 import triton
diff --git a/python/examples/test_mm.py b/python/examples/test_mm.py
index 5d5ba57..5bbdcd5 100644
--- a/python/examples/test_mm.py
+++ b/python/examples/test_mm.py
@@ -1,6 +1,3 @@
-# Copyright (c) Meta Platforms, Inc. and affiliates, Microsoft Corporation.
-# Licensed under the MIT license.
-
 
 import torch
 import triton
diff --git a/python/examples/test_modulo.py b/python/examples/test_modulo.py
index 130618b..f7e1775 100644
--- a/python/examples/test_modulo.py
+++ b/python/examples/test_modulo.py
@@ -1,6 +1,3 @@
-# Copyright (c) Meta Platforms, Inc. and affiliates, Microsoft Corporation.
-# Licensed under the MIT license.
-
 import torch
 
 import triton
diff --git a/python/examples/test_nested_loops.py b/python/examples/test_nested_loops.py
index 5ff3a79..9ce055a 100644
--- a/python/examples/test_nested_loops.py
+++ b/python/examples/test_nested_loops.py
@@ -1,6 +1,3 @@
-# Copyright (c) Meta Platforms, Inc. and affiliates, Microsoft Corporation.
-# Licensed under the MIT license.
-
 import torch
 
 import triton
diff --git a/python/examples/test_reduce.py b/python/examples/test_reduce.py
index 3f7339f..2b46a83 100644
--- a/python/examples/test_reduce.py
+++ b/python/examples/test_reduce.py
@@ -1,6 +1,3 @@
-# Copyright (c) Meta Platforms, Inc. and affiliates, Microsoft Corporation.
-# Licensed under the MIT license.
-
 import torch
 import pytest
 import math
diff --git a/python/examples/test_scalar_store.py b/python/examples/test_scalar_store.py
index 4249a92..7ef14d4 100644
--- a/python/examples/test_scalar_store.py
+++ b/python/examples/test_scalar_store.py
@@ -1,6 +1,3 @@
-# Copyright (c) Meta Platforms, Inc. and affiliates, Microsoft Corporation.
-# Licensed under the MIT license.
-
 import torch
 
 import triton
diff --git a/python/examples/test_sign_extend.py b/python/examples/test_sign_extend.py
index 50928b7..21726e7 100644
--- a/python/examples/test_sign_extend.py
+++ b/python/examples/test_sign_extend.py
@@ -1,6 +1,3 @@
-# Copyright (c) Meta Platforms, Inc. and affiliates, Microsoft Corporation.
-# Licensed under the MIT license.
-
 import torch
 
 import triton
diff --git a/python/examples/test_softmax.py b/python/examples/test_softmax.py
index 9bf4257..b3d43df 100644
--- a/python/examples/test_softmax.py
+++ b/python/examples/test_softmax.py
@@ -1,6 +1,3 @@
-# Copyright (c) Meta Platforms, Inc. and affiliates, Microsoft Corporation.
-# Licensed under the MIT license.
-
 import torch
 
 import triton
diff --git a/python/examples/test_splat.py b/python/examples/test_splat.py
index dd79e15..a396b5e 100644
--- a/python/examples/test_splat.py
+++ b/python/examples/test_splat.py
@@ -1,6 +1,3 @@
-# Copyright (c) Meta Platforms, Inc. and affiliates, Microsoft Corporation.
-# Licensed under the MIT license.
-
 import torch
 
 import triton
diff --git a/python/examples/test_swap.py b/python/examples/test_swap.py
index 5a6a7a1..2693ab6 100644
--- a/python/examples/test_swap.py
+++ b/python/examples/test_swap.py
@@ -1,6 +1,3 @@
-# Copyright (c) Meta Platforms, Inc. and affiliates, Microsoft Corporation.
-# Licensed under the MIT license.
-
 import torch
 
 import triton
diff --git a/python/examples/test_tensor_index_iterargs.py b/python/examples/test_tensor_index_iterargs.py
index 159b9d3..4b3ac80 100644
--- a/python/examples/test_tensor_index_iterargs.py
+++ b/python/examples/test_tensor_index_iterargs.py
@@ -1,6 +1,3 @@
-# Copyright (c) Meta Platforms, Inc. and affiliates, Microsoft Corporation.
-# Licensed under the MIT license.
-
 import torch
 
 import triton
diff --git a/python/examples/test_unstructured_mask.py b/python/examples/test_unstructured_mask.py
index 58302cc..2b37c2a 100644
--- a/python/examples/test_unstructured_mask.py
+++ b/python/examples/test_unstructured_mask.py
@@ -1,6 +1,3 @@
-# Copyright (c) Meta Platforms, Inc. and affiliates, Microsoft Corporation.
-# Licensed under the MIT license.
-
 import torch
 
 import triton
diff --git a/python/examples/test_vec_add.py b/python/examples/test_vec_add.py
index 49e92c6..db2fa09 100644
--- a/python/examples/test_vec_add.py
+++ b/python/examples/test_vec_add.py
@@ -1,6 +1,3 @@
-# Copyright (c) Meta Platforms, Inc. and affiliates, Microsoft Corporation.
-# Licensed under the MIT license.
-
 import torch
 
 import triton
diff --git a/test/lit.cfg.py b/test/lit.cfg.py
index 54a7d16..72c9cc7 100644
--- a/test/lit.cfg.py
+++ b/test/lit.cfg.py
@@ -1,7 +1,4 @@
 # -*- Python -*-
-# Copyright (c) Meta Platforms, Inc. and affiliates, Microsoft Corporation.
-# Licensed under the MIT license.
-
 
 import os
 import platform
diff --git a/tools/triton-shared-opt/RegisterTritonSharedDialects.h b/tools/triton-shared-opt/RegisterTritonSharedDialects.h
index 0165b03..d78f96b 100644
--- a/tools/triton-shared-opt/RegisterTritonSharedDialects.h
+++ b/tools/triton-shared-opt/RegisterTritonSharedDialects.h
@@ -1,10 +1,3 @@
-//===----------------------------------------------------------------------===//
-//
-// Copyright (c) Meta Platforms, Inc. and affiliates, Microsoft Corporation.
-// Licensed under the MIT license.
-//
-//===----------------------------------------------------------------------===//
-
 #pragma once
 #include "mlir/Dialect/Bufferization/IR/Bufferization.h"
 #include "mlir/Dialect/Func/IR/FuncOps.h"
diff --git a/tools/triton-shared-opt/triton-shared-opt.cpp b/tools/triton-shared-opt/triton-shared-opt.cpp
index ff19e0e..3ed831d 100644
--- a/tools/triton-shared-opt/triton-shared-opt.cpp
+++ b/tools/triton-shared-opt/triton-shared-opt.cpp
@@ -1,6 +1,6 @@
 //===----------------------------------------------------------------------===//
 //
-// Copyright (c) Meta Platforms, Inc. and affiliates, Microsoft Corporation.
+// Copyright (c) Microsoft Corporation.
 // Licensed under the MIT license.
 //
 //===----------------------------------------------------------------------===//
diff --git a/triton-hash.txt b/triton-hash.txt
index 57a17d5..ca05e91 100644
--- a/triton-hash.txt
+++ b/triton-hash.txt
@@ -1 +1 @@
-32b7b208d48d2d06c8af2cbf1b45d8c4c70d7f1f
\ No newline at end of file
+32b7b208d48d2d06c8af2cbf1b45d8c4c70d7f1f
diff --git a/triton-san/examples/buffer-overflow.py b/triton-san/examples/buffer-overflow.py
index 4f647fb..a0599a0 100644
--- a/triton-san/examples/buffer-overflow.py
+++ b/triton-san/examples/buffer-overflow.py
@@ -1,6 +1,3 @@
-# Copyright (c) Meta Platforms, Inc. and affiliates, Microsoft Corporation.
-# Licensed under the MIT license.
-
 import torch
 import triton
 import triton.language as tl
diff --git a/triton-san/examples/data-race.py b/triton-san/examples/data-race.py
index 86a0fed..cfca91b 100644
--- a/triton-san/examples/data-race.py
+++ b/triton-san/examples/data-race.py
@@ -1,6 +1,3 @@
-# Copyright (c) Meta Platforms, Inc. and affiliates, Microsoft Corporation.
-# Licensed under the MIT license.
-
 import torch
 import triton
 import triton.language as tl
diff --git a/triton_shared.cc b/triton_shared.cc
index 43058b6..e69ed1d 100644
--- a/triton_shared.cc
+++ b/triton_shared.cc
@@ -1,11 +1,4 @@
-﻿//===----------------------------------------------------------------------===//
-//
-// Copyright (c) Meta Platforms, Inc. and affiliates, Microsoft Corporation.
-// Licensed under the MIT license.
-//
-//===----------------------------------------------------------------------===//
-
-#include <pybind11/pybind11.h>
+﻿#include <pybind11/pybind11.h>
 
 namespace py = pybind11;
 
